src/charm.py

```diff
@@ -116,7 +122,20 @@ def _agent_conf(self, key: str):
 
     def api_port(self) -> str:
         """Return the port on which the controller API server is listening."""
-        return self._agent_conf('apiport')
+        api_port = self._agent_conf('apiport')
```

Comment:
I'd just get rid of the apiport parsing. apiaddresses is the correct way to handle this as apiport is only a concern of the controller agent not the controller unit agent.